python/cudf/cudf/core/multiindex.py

```diff
@@ -775,7 +775,10 @@ def _index_and_downcast(self, result, index, index_key):
             )
 
         if isinstance(index_key, tuple):
-            result = result.set_index(index)
+            if isinstance(result, cudf.Series):
+                result.index = index
+            else:
+                result = result.set_index(index)
```

Comment:
At this point it is safe to assume that index is always an index object, so I don't think you need this branching. 
```suggestion
            result.index = index
```

Reply:
@vyasr This is not branching on the type of `index`. It's branching on the type of `result`. If `result` is a `Series`, we set its `index` attribute. If `result` is a `DataFrame`, we call `DataFrame.set_index(index)`. I don't think `DataFrame.set_index` does the same thing as setting its `index` attribute.

Reply:
I tested this change locally and I see that `test_memory_usage_multi` fails with this change. It gets a size of 368 instead of 360. I take this to mean that there is some difference between calling `DataFrame.set_index` and setting the `DataFrame.index` attribute.

Reply:
I chatted offline with @bdice about this. The failing test indicates to me that there is indeed a difference between the two, but not one that should be present. Fixing that is out of scope for this PR, so I've asked him to add an appropriate comment here and we can move forward with the code as is for now.

Reply:
Comment added in 9ca4b4cf23.

Reply:
@bdice sorry for turning this around again. I just tried to make this change locally using your branch and I find the failure to be extremely flaky. I observe the failure both with and without this change, at random intervals. You can use pytest-repeat with `pytest --count=100 python/cudf/cudf/tests/test_dataframe.py::test_memory_usage_multi` to hopefully see this for yourself. The failure is consistently that the computed memory usage underestimates the expected by 8 bytes (`368 != 360`) on a MultiIndex with 10 rows. I am investigating this further because I'm curious what the cause of this discrepancy is, but I think it means that you can safely make this change (and remove the TODO) in this PR.

Reply:
The memory test issue is resolved in #10114. This change can be made safely in this PR.